Film, User. FilmController, UserController.

nixanok opened the pull request:
Логика добавления фильма и юзера. Валидация данных.
Реализованы тесты. Логирование.

nixanok committed:
Хранилища Film и User.
FilmController и UserController.
Базовые методы добавления, обновления и получения.

nixanok committed:
Delete mvnw

nixanok committed:
Delete mvnw.cmd

nixanok committed:
Удалены неиспользованные импорты.

nixanok committed:
Код приведен к чекстайлу.

nixanok committed:
Код приведен к чекстайлу.

nixanok committed:
<opt>:Исправлены ошибки с отсутсвием модификатора доступа. Добавлены тесты. Оптимизирована работа запросов при невыполненной валидации данных.

nixanok committed:
checkStyle

nixanok committed:
checkStyle

nixanok committed:
checkStyle

nixanok committed:
checkStyle

nixanok committed:
checkStyle

AlexeyCheuzov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,91 @@
+            ResponseError error = ResponseError
+                    .builder()
+                    .timestamp(LocalDateTime.now())
+                    .status(500)
```
В данном случае лучше возвращать например 404 ошибку, так как 500 это скорей ошибка сервера, но то что фильм не найден, может быть ошибкой пользователя, который пытается загрузить некорректный фильм.

AlexeyCheuzov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,25 @@
+    @NotBlank(message = "Email cannot be blank")
+    private String email;
+
+    @Past(message = "birthday should be in the past")
```
Упустил момент в предыдущей итерации. Вероятность того, что день рождения пользователя сегодня, крайне мала, но не равна нулю) Тем более, в тз указано, что день рождения пользователя не может быть в будущем, поэтому логичнее поставить аннотацию PastOrPresent

nixanok committed:
<fix>:"Birthday user now can be in present."
<fix>:"If the user is not found, the response code is set to 404."
<opt>:"Added displayNames to tests."
